### PR TITLE
[WIP] Using ILogger instead of EventSource for logging OpenTelemetry Events in OTLP exporter

### PR DESCRIPTION
**Pull request by Yun-Ting**
Towards https://github.com/open-telemetry/opentelemetry-dotnet/issues/3881

## Changes

Currently internal logging with ILogger is default on. 
Need to find a way to have the internal diagnostics off by default and a switch to turn the diagnostics on.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpLogExporter.cs [hunk 1]
```diff
@@ -75,6 +88,9 @@ public override ExportResult Export(in Batch<LogRecord> logRecordBatch)
         catch (Exception ex)
         {
             OpenTelemetryProtocolExporterEventSource.Log.ExportMethodException(ex);
+
+            OpenTelemetryProtocolExporterEvents.ExportMethodException(this.openTelemetryEventLogger, ex.ToInvariantString());
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpLogExporter.cs:92]: we just need to pass exception object itself to logger. It won't get serialized, unless someone is listening to that. If someone is listening, the providers know how to serialize exception into string representation, where EventSource required raw types like string, and can't work with types like `Exception`

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEvents.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
+
+/// <summary>
+/// Methods for logging internal events in OpenTelemetryProtocolExporter.
+/// </summary>
+internal static partial class OpenTelemetryProtocolExporterEvents
+{
+    // TODO: there's no concept of Event vs NonEvent in ILogger (as opposed to EventSource.)
+    // Need to find a way to log "NonEvent" in ILogger to avoid the below situation:
+    // When the Exception occurred in ExportMethod, the Export event got written to the pipe, and thus the Exporter keeps exporting.
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEvents.cs:15]: SupressInstrumentationScope should be preventing infinite loop already.. is that what you are referring to there?
export() -> ilooger.log(exception) -> gets-back-to-export export() -> ilogger.log(exception) -- this infinite cycle?

File: examples/AspNetCore/Program.cs [hunk 3]
```diff
@@ -29,7 +29,8 @@
 appBuilder.Services.AddSingleton<Instrumentation>();
 
 // Clear default logging providers used by WebApplication host.
```
[the remark this answers is not in the document]
Comment (Yun-Ting) [reply examples/AspNetCore/Program.cs:31]: The change in the example app is just for local testing purposes while I'm working on the integration tests for this.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpLogExporter.cs [hunk 4]
```diff
@@ -26,27 +30,34 @@ public sealed class OtlpLogExporter : BaseExporter<LogRecord>
...
         OtlpExporterTransmissionHandler<OtlpCollector.ExportLogsServiceRequest>? transmissionHandler = null)
     {
+        Debug.Assert(openTelemetryEventLogger != null, "openTelemetryEventLogger was null");
         Debug.Assert(exporterOptions != null, "exporterOptions was null");
         Debug.Assert(sdkLimitOptions != null, "sdkLimitOptions was null");
         Debug.Assert(experimentalOptions != null, "experimentalOptions was null");
 
+        this.openTelemetryEventLogger = openTelemetryEventLogger!;
+
+        this.openTelemetryEventLogger.LogDebug("Hello from Otlp ctor");
```
[the remark this answers is not in the document]
Comment (Yun-Ting) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpLogExporter.cs:59]: Yes, `.LogDebug()` is for debugging purposes and will be removed in the upcoming iterations.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEvents.cs (hunk 2 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEvents.cs:15]: Yes, the above call sequence is what I'm referring to. Let me see how SuppressInstrumentationScope can prevent this from happening. Thanks!

**Pull request closed without merging**